src/test/java/pl/edu/icm/pl/mxrdr/extension/workflow/XdsImagesPatternStepTest.java

```diff
@@ -0,0 +1,39 @@
+package pl.edu.icm.pl.mxrdr.extension.workflow;
+
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.ArgumentsProvider;
+import org.junit.jupiter.params.provider.ArgumentsSource;
+
+import java.util.stream.Stream;
+
+import static java.util.Collections.singletonMap;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static pl.edu.icm.pl.mxrdr.extension.workflow.XdsImagesPatternStep.FILE_NAMES_PARAM_NAME;
+import static pl.edu.icm.pl.mxrdr.extension.workflow.XdsImagesPatternStep.FILE_NAMES_SEPARATOR;
+
+class XdsImagesPatternStepTest implements ArgumentsProvider {
+
+    @ParameterizedTest(name = "[{index}] \"{1}\" pattern")
+    @ArgumentsSource(XdsImagesPatternStepTest.class)
+    void shouldCalculatePatterns(String names, String expected) {
+        XdsImagesPatternStep step = new XdsImagesPatternStep(null, singletonMap(FILE_NAMES_PARAM_NAME, names));
```

Comment:
I usually do that with labels when using Spock. Otherwise try to keep the test short by extracting methods and classes. When each section has a single line it's quite simple to identify each of them. Therefore I only simply put an empty line between them, as they always come in the same order in every test.